flang/lib/Optimizer/Builder/HLFIRTools.cpp

```diff
@@ -860,7 +859,9 @@ translateVariableToExtendedValue(mlir::Location loc, fir::FirOpBuilder &builder,
                                 fir::MutableProperties{});
 
   if (base.getType().isa<fir::BaseBoxType>()) {
-    if (!variable.isSimplyContiguous() || variable.isPolymorphic() ||
+
+    bool contiguous = variable.isSimplyContiguous() || contiguousHint;
```

Comment:
I am not sure about this change yet to be honest. In particular, I am testing the effect of changing this line to: `bool contiguous = variable.isSimplyContiguous() || variable.isArray();`.

@jeanPerier let me know if you think the above change is a bad idea for any reason.

Reply:
variable.isArray() would not be correct since this would be true for any array, including `x(:)` that is not contiguous. Your change using the extra knowledge you have from the symbol/original box makes sense to me.